### ISSUE TITLE
Ternary expression with @MonotonicNonNull fails

### ISSUE THREAD

**Issue by elcelista** [typetools/checker-framework#1256]
I would expect the following to check successfully, with `bar` receiving the inferred type `@Nullable Object`:

```
import org.checkerframework.checker.nullness.qual.*;

public class MonotonicNonNullTernaryIf {

  @MonotonicNonNull Object defaultFoo;

  public void test(Object foo, boolean condition) {
    Object bar = condition ? foo : defaultFoo;
  }
}
```

But instead I get the following error:

```
$ java -jar checker.jar -processor org.checkerframework.checker.nullness.NullnessChecker -version MonotonicNonNullTernaryIf.java 
javac 1.8.0-jsr308-2.1.10
MonotonicNonNullTernaryIf.java:8: error: [monotonic.type.incompatible] cannot assign org.checkerframework.checker.nullness.qual.MonotonicNonNull to org.checkerframework.checker.nullness.qual.MonotonicNonNull (monotonic type).
    Object bar = condition ? foo : defaultFoo;
                                   ^
  full type found: @Initialized @MonotonicNonNull Object
1 error
```

**Comment by kanha95** [comment]
It's similar to Issue #1170.

```java
import org.checkerframework.checker.nullness.qual.*;

public class MonotonicNonNullTernaryIf {

  @MonotonicNonNull Object defaultFoo;

  public void test(Object foo, boolean condition) {
    Object bar;
    if(condition) {
        bar=foo;
    }
    else {
    bar=defaultFoo;
      }    
   }
}

```
It passes using if-else statement but fails in case of ternary operators.

**Comment by mernst** [comment]
A workaround is to use `if ... else`.  This is undesirable, but will work until we fix this issue.
